### ISSUE TITLE
Pretraining SciBERT

### ISSUE THREAD

**Issue by xycforgithub** [allenai/scibert#32]
Hi,
The repo does not seem to contain the codes to pretrain the model on semantic scholar. Do you plan to release those codes and the pretrain data? Thanks!

Yichong

**Comment by ibeltagy** [comment]
We used the BERT code from google https://github.com/google-research/bert

**Comment by xycforgithub** [comment]
Thanks! And where did you get the data?

**Comment by ibeltagy** [comment]
As mentioned in the paper, we used the semantic scholar corpus which is not publicly available. The publicly available part is this https://labs.semanticscholar.org/corpus/ (the link is not working right now but it will be fixed soon), which has titles and abstracts but not full text.

**Comment by eric-haibin-lin** [comment]
@ibeltagy thanks for the reply. What mlm_loss should I expect the model to converge to if I used the same dataset?

**Comment by ibeltagy** [comment]
With 512 tokens, the losses are around the following numbers:
```global_step = 800000
loss = 1.311045
masked_lm_accuracy = 0.7187241
masked_lm_loss = 1.2882848
next_sentence_accuracy = 0.9939219
next_sentence_loss = 0.0196654```

**Comment by sibyl1956** [comment]
@ibeltagy Thanks for reply. Does that mean that your pretrained Sci-Bert model reaches masked_lm_accuracy of around 0.718? However, in the BERT original model, they reaches around 0.98 masked_lm_accuracy and about 1.0 next_sentence accuracy. May I ask do you think a masked_lm_accuracy of around 0.718 is enough? 
I am training also my own model on a customized dataset, which adds around 1000 new tokens that are not in BERT model. Thus I would like to know what is the masked_lm_accuracy or next_sentence_accuracy that I should expect for my pre-trained model to achieve.

**Comment by kyleclo** [comment]
@sibyl1956  This is a good point and I suspect this is due to the noisy PDF parse in the scientific corpus.  Namely, we didn't do anything to remove tables, equations, weird tokens, etc. that was output by PDFBox in converting the raw PDF to a text stream.  You can expect it's essentially impossible for the model to really predict these masked tokens.  We're currently in the process of curating an updated larger & cleaner version of the pretraining corpus, and will investigate whether the noisy tokens are the cause for this.  As it is now, the current released SciBERT weights are still very good for downstream tasks (but we can definitely do better).

**Comment by kyleclo** [comment]
I'm closing this issue for now since it looks like the original question chain was answered.  Feel free to reopen or start new Issue
